fix(sensor): remove dead image cache

janisz opened the pull request:
### Description

This PR removes image cache usage from sensor as it always results with miss due to a bad expected type. Here we try to get `*storage.Image` while the only values are `cacheValue`. As type assertion fails, image is always `nil` which result in a miss.

There are only 2 places that adds to image cache:

https://github.com/stackrox/stackrox/blob/7d64f4eb4e1eecac3fcd8f09bbd31266d0b12910/sensor/common/detector/enricher.go#L292

https://github.com/stackrox/stackrox/blob/7d64f4eb4e1eecac3fcd8f09bbd31266d0b12910/sensor/common/detector/detector.go#L337

I discovered this issue while working on adding a generic types to a cache
- https://github.com/stackrox/stackrox/pull/11925#discussion_r1673965036
- https://github.com/stackrox/stackrox/pull/11924
---
- [x] CHANGELOG update is not needed
- [x] Documentation is not needed
### Testing
- [x] inspected CI results
#### Automated testing
- [x] modified existing tests
- [x] contributed **no automated tests**
#### How I validated my change
CI

dcaravel reviewed (commented):
What was the motivation for removal vs. fixing?

This service is used by the admission controller - the calls are time sensitive - even though each admission controller has a separate cache the 'semi-centralized' cache here in sensor would prevent unnecessary re-scans in various scenarios.

janisz commented:
> What was the motivation for removal vs. fixing?

Good question. I've tried to fix it by using cschevalue every where. This leads to e2e failures. So, since I don't have context for this code I decided to remove — it's not working anyway.

lvalerom commented on sensor/common/image/service_impl.go:
```diff
@@ -67,16 +63,6 @@ func (s *serviceImpl) SetClient(conn grpc.ClientConnInterface) {
 func (s *serviceImpl) GetImage(ctx context.Context, req *sensor.GetImageRequest) (*sensor.GetImageResponse, error) {
-	if id := req.GetImage().GetId(); id != "" {
-		v, _ := s.imageCache.Get(imagecacheutils.GetImageCacheKey(req.GetImage()))
-		img, _ := v.(*storage.Image)
```
What was the problem with the fix you did in your other PR?
```golang
	id := req.GetImage().GetId()
	v, ok := s.imageCache.Get(imagecacheutils.GetImageCacheKey(req.GetImage()))
	if id != "" && ok {
		img := v.WaitAndGet()
```

janisz replied:
Tests are failing :(
I can spent some time to debug it but I have not much context why maybe they wait forever :thinking: 
Maybe we need another cache that will just store images without this wait logic

lvalerom replied:
I think that having another cache with only the images could solve the issue but we will have to sync the two caches and that could lead to other complications. I think it's worth trying to understand the problem and fix it. Which tests in particular are failing because of this? I saw a lot of tests failing in the other PR, but I'm not sure which ones are related to this.

Interestingly, I found something that, while I think it works, it might work by accident 😆 . [Here](https://github.com/stackrox/stackrox/blob/7d64f4eb4e1eecac3fcd8f09bbd31266d0b12910/sensor/common/detector/detector.go#L329-L340) we create a new `cacheValue` but we do not create the signal, nor do we trigger it. I think this works because the default state of a `Signal` is 'triggered' so that `cacheValue` will not block if `WaitAndGet` is called, but I think this is by accident and not really intentional.

janisz commented:
@lvalerom Thank you for a hint with signals. I've updated the other PR to fix this cache and use image only if it was scanned so no need to wait. I added a method `GetIfDone()` that achieves that.
